transit_odp/browse/views/local_authority.py

```diff
@@ -142,49 +143,41 @@ def get_context_data(self, **kwargs):
         ids_list = {}
 
         for lta in all_ltas_current_page:
-            if lta.ui_lta_name_trimmed not in ids_list:
-                ids_list[lta.ui_lta_name_trimmed] = [lta.id]
-            else:
-                ids_list[lta.ui_lta_name_trimmed].append(lta.id)
-
-        for lta_id_list in ids_list.values():
-            for lta in all_ltas_current_page:
-                if lta.id in lta_id_list:
-                    setattr(lta, "auth_ids", lta_id_list)
+            lta_list = (
+                self.model.objects.annotate(ui_lta_name_trimmed=Trim("ui_lta_name"))
+                .filter(ui_lta_name_trimmed=lta.ui_lta_name_trimmed)
+                .all()
+            )
```

Comment:
Can we run the query once to obtain all the ltas outside the loop instead of invoking a db call within the for loop? We could re-use `all_ltas` but instead of using the queryset we use the model objects.